clusterloader2/pkg/ticker/timer.go

```diff
@@ -17,18 +17,54 @@ limitations under the License.
 package ticker
 
 import (
+	"fmt"
+	"math/rand"
 	"time"
 
 	"k8s.io/perf-tests/clusterloader2/api"
 )
 
-func newTimer(ch chan<- struct{}, ts *api.TuningSet) *time.Timer {
+func newTimer(ch chan<- struct{}, ts *api.TuningSet) (*time.Timer, error) {
 	var t *time.Timer
+	getTicks, err := getTicksFunc(ts)
+	if err != nil {
+		return nil, err
+	}
+	getDelay, err := getDelayFunc(ts)
+	if err != nil {
+		return nil, err
+	}
 	t = time.AfterFunc(ts.InitialDelay, func() {
-		// TODO: Implement other tuning sets.
-		d := 10 * time.Millisecond
-		ch <- struct{}{}
-		t.Reset(d)
+		for i := int32(0); i < getTicks(); i++ {
+			ch <- struct{}{}
+		}
+		t.Reset(getDelay())
 	})
-	return t
+	return t, nil
+}
+
+func getTicksFunc(ts *api.TuningSet) (func() int32, error) {
+	if ts.QpsLoad != nil || ts.RandomizedLoad != nil {
+		return func() int32 { return int32(1) }, nil
+	}
+	if ts.SteppedLoad != nil {
+		return func() int32 { return ts.SteppedLoad.BurstSize }, nil
+	}
+	return nil, fmt.Errorf("unknown tuningset structure: %v", ts)
+}
+
+func getDelayFunc(ts *api.TuningSet) (func() time.Duration, error) {
```

Comment:
Please add a general comment above explaining what this function is calculating. Or even better if you add a comment above each mode explaining why you compute the delay the way you do it.

Reply:
Comments added.